js/utils.js

```diff
@@ -2102,6 +2110,50 @@ Fliplet.Registry.set('dynamicListUtils', (function() {
     });
   }
 
+  /**
+   * @param {Object} linkAction - link provider result object with required field `page`
+   * @param {String} queryParams - default query parameters that we want to add when navigating to another screen
+   *
+   * @returns {Promise} returns a promise of the Fliplet.Navigate.to method.
+   */
+  function navigateToTheScreen(linkAction, queryParams) {
+    if (!_.get(linkAction, 'page')) {
+      return Promise.reject('Page error');
+    }
+
+    linkAction.query = appendUrlQuery(linkAction.query, queryParams);
+
+    return Fliplet.Navigate.to(linkAction);
+  }
+
+  /**
+   *
+   * @param {String} error - error message from the navigateToTheScreen function
+   * @param {Object} errorMessages - Messages that we will show in toast.
+   *  required poperties:
+   *    'pageError' - message that we will show when no page is specifyed
+   *    'openError' - message that we will show when error ocures on page open
+   *
+   * @returns {void}
+   */
+  function navigateToScreenErrorHandler(error, errorMessages) {
+    switch (error) {
+      case 'Page error':
+        Fliplet.UI.Toast({
+          title: 'Link not configured',
+          message: errorMessages.pageError
+        });
+
+        break;
+      default:
+        Fliplet.UI.Toast(error, {
```

Comment:
This tells me you hadn't tested this case.

```suggestion
        Fliplet.UI.Toast.error(error, {
```